site-scrapers/TrinityEMS/index.js

```diff
@@ -0,0 +1,253 @@
+                </td>
+                */
+                const date = await page.evaluate(
+                    (el) => el.getAttribute("day"),
+                    day
+                );
+
+                const slotCount = await getSlotsForDate(page, date);
+
+                monthlyAvailability[reformatDate(date)] = {
+                    hasAvailability: !!slotCount,
+                    numberAvailableAppointments: slotCount,
+                };
+            }
+        } catch (error) {
+            console.log(
+                `${site.name} :: error trying to get day numbers: ${error}`
+            );
+        }
+    }
+
+    return monthlyAvailability;
+}
+
+/**
+ * Looks for "no-dates-available" class, or ".activeday" elements.
+ * If not found, there are active days, and the page should be
+ * parsed for which days have availability.
+ *
+ * @returns Array of active day elements. Empty array is returned if there's no availability.
+ */
+async function getActiveDays(page) {
+    const noDatesAvailable = await page.$(".no-dates-available");
+    // No appointments available if defined
+    if (noDatesAvailable != undefined) {
+        return [];
+    } else {
+        let days = [];
+        try {
+            days = await page.$$(".activeday");
+        } catch (error) {
+            console.log(`error trying to get day numbers: ${error}`);
+        }
+        return days;
+    }
+}
+
+/**
+ * Fetches the small snippet of HTML which would appear in a popup when an active day is clicked by a human.
+ *
+ * As yet, it's not know how the slot count will be presented. See parseHTMLforSlotCount().
+ * @param {*} page
+ * @param {String} dateStr -- yyyy-mm-dd, as in '2021-04-26'
+ * @returns response text of available times XHR query
+ */
+async function getSlotsForDate(page, dateStr) {
+    const url = [
+        "https://app.acuityscheduling.com/schedule.php?action=availableTimes",
+        "showSelect=0",
+        "fulldate=1",
+        "owner=21713854",
+        "type=19620839",
+        "calendar=5109380",
+        `date=${dateStr}`,
+        "ignoreAppointment=",
+    ].join("&");
+
+    return await page.evaluate(async (url) => {
+        const response = await fetch(url);
+        const text = await response.text();
+        return parseHTMLforSlotCount(text);
+    }, url);
+}
+
+/**
+ * Advances the page to the next month by clicking on the "next" button.
+ *
+ * @param {*} page
+ */
+async function advanceMonth(page) {
+    const nextMonthLinkClass = ".calendar-next";
+
+    try {
+        await Promise.all([
+            await page.click(nextMonthLinkClass),
+            // If the timeout option is not set, the timeout (30000 ms) will expire, and
+            // no results will be returned. Setting it to something shorter yields results.
+            // 5000 seems to be too much because the pages return nearly instantaneously.
+            // But will leave it at that just in case of ... what?
```

Comment:
Can you finish out this thought? Leaving it ambiguous is a bit confusing!

Reply:
It is confusing, both the comment and the issue of why it times out at the default 30000 ms, but works with shorter time outs. Edited the comment; in next commit.